plugins/inputs/t128_tank/t128_tank.go

```diff
@@ -0,0 +1,84 @@
+package t128_tank
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+var cliPath = "/opt/128technology/bin/tank-cli"
+var topic = "events"
+
+var sampleConfig = `
+[[inputs.t128_tank]]
+## Required. A name for the collector which will be used as the measurement name of the produced data.
+# collector_name = "event_collector"
+
+## Required.
+# log_name = "event_collector"
+
+## Required.  
+# index_file = ""
+
+## Required.
+# topic = "events"
+`
+
+type T128Tank struct {
+	CollectorName string `toml:"collector_name"`
+	LogName       string `toml:"log_name"`
+	IndexFile     string `toml:"index_file"`
+	Topic         string `toml:"topic"`
+}
+
+func (*T128Tank) SampleConfig() string {
+	return sampleConfig
+}
+
+func (*T128Tank) Description() string {
+	return "Make a 128T TANK query and return the data"
+}
+
+func (plugin *T128Tank) Init() error {
+	return nil
+}
+
+func (plugin *T128Tank) Gather(acc telegraf.Accumulator) error {
```

Comment:
This will be a long running input called a [Service Input](https://github.com/128technology/telegraf/blob/master/docs/INPUTS.md#service-input-plugins). It runs a little differently and doesn't have discrete gather times. Check out the docs and the `execd` input.